packages/@aws-cdk/aws-stepfunctions-tasks/lib/publish-to-topic.ts

```diff
@@ -0,0 +1,81 @@
+import cloudwatch = require('@aws-cdk/aws-cloudwatch');
+import iam = require('@aws-cdk/aws-iam');
+import sns = require('@aws-cdk/aws-sns');
+import sfn = require('@aws-cdk/aws-stepfunctions');
+import cdk = require('@aws-cdk/cdk');
+import { renderString } from './json-path';
+
+/**
+ * Properties for PublishTask
+ */
+export interface PublishToTopicProps {
+  /**
+   * The text message to send to the queue.
+   *
+   * Exactly one of `message` and `messageObject` is required.
+   */
+  readonly message?: string;
+
+  /**
+   * Object to be JSON-encoded and used as message
+   *
+   * Exactly one of `message`, `messageObject` and `messagePath` is required.
+   */
+  readonly messageObject?: string;
+
+  /**
+   * If true, send a different message to every subscription type
+   *
+   * If this is set to true, message must be a JSON object with a
+   * "default" key and a key for every subscription type (such as "sqs",
+   * "email", etc.) The values are strings representing the messages
+   * being sent to every subscription type.
+   *
+   * @see https://docs.aws.amazon.com/sns/latest/api/API_Publish.html#API_Publish_RequestParameters
+   */
+  readonly messagePerSubscriptionType?: boolean;
+
+  /**
+   * Message subject
+   */
+  readonly subject?: string;
+}
+
+/**
+ * A StepFunctions Task to invoke a Lambda function.
+ *
+ * A Function can be used directly as a Resource, but this class mirrors
+ * integration with other AWS services via a specific class instance.
+ */
+export class PublishToTopic implements sfn.IStepFunctionsTask {
+  public readonly resourceArn: string;
```

Comment:
Something feels wrong about exposing all of these as public API. I'd expected that `IStepFunctionTask` would have a single method `toTaskProperties` method that returns all of these, and then we don't need to pollute the public API with the low-level details.

Reply:
I had that first, but I honestly find that more complicated and the benefit seemed fringy. We used to have that because the API surface area of the resource class was shared with things users were expected to interact directly with, whereas now the class only has a single purpose (be the integration class), so there's no other surface area to interact with.

Reply:
I don’t agree that the benefit is fringy. The integration class should encapsulate those details and not leak them to the user. There could be APIs that the integration class exposes and then that will become messy. 

It’s also more consistent with other existing integration classes like event targets.

Reply:
In fact I think these methods should be called “bind” and they should accept the source construct so they get an opportunity to bind to the tree